biahub/segment.py

```diff
@@ -236,12 +269,12 @@ def segment_cli(
             jobs.append(
                 executor.submit(
                     process_single_position,
-                    segment_data,
-                    input_position_path,
-                    output_position_path,
+                    func=segment_data,
+                    input_position_path=input_position_path,
+                    output_position_path=output_position_path,
                     input_channel_indices=[list(range(C))],
                     output_channel_indices=[list(range(C_segment))],
-                    num_processes=np.min([20, int(num_cpus * 0.8)]),
+                    num_processes=np.min([slurm_array_parallelism, int(num_cpus * 0.8)]),
```

Comment:
The number of processes here should be decoupled from the `slurm_array_parallelism`?

Reply:
It's tricky because this depends on the input dataset.